### PR TITLE
fix: notification services add new allowed events

### PR DESCRIPTION
**Pull request by matteoscurati**
## **Description**

This PR introduces a change to the `metamask-controller.js` file to add an allowed action to the list of permitted actions for the `NotificationServicesController`. Specifically, the action `subscribeToPushNotifications` has been added for the `NotificationServicesPushController`.

This PR is necessary after the release of this update: https://github.com/MetaMask/core/pull/4653

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26987?quickstart=1)

## **Related issues**

N/A

## **Manual testing steps**

N/A

## **Screenshots/Recordings**

N/A

### **Before**

N/A

### **After**

N/A

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/scripts/metamask-controller.js [hunk 1]
```diff
@@ -1485,7 +1485,12 @@ export default class MetamaskController extends EventEmitter {
           'NotificationServicesController:disableNotificationServices',
           'NotificationServicesController:selectIsNotificationServicesEnabled',
         ],
-        allowedEvents: ['KeyringController:lock', 'KeyringController:unlock'],
+        allowedEvents: [
+          'KeyringController:lock',
+          'KeyringController:unlock',
+          'AccountsController:accountAdded',
```
Comment (mathieuartu) [inline app/scripts/metamask-controller.js:1491]: Oh you had to add those! Wasn't sure you would need to before I activated account syncing. Thanks!

**Review by mathieuartu — DISMISSED** [review]
LGTM!

File: app/scripts/metamask-controller.js (hunk 1 above)
Comment (Prithpal-Sooriya) [reply app/scripts/metamask-controller.js:1491]: Yep test oddly expect the correct allowed events need to be provided.
All good. I am concerned that this controller will contain many events in the future - we can worry/split this out later though.

File: app/scripts/metamask-controller.js [hunk 2]
```diff
@@ -1560,6 +1565,7 @@ export default class MetamaskController extends EventEmitter {
             'UserStorageController:performSetStorage',
             'NotificationServicesPushController:enablePushNotifications',
             'NotificationServicesPushController:disablePushNotifications',
+            'NotificationServicesPushController:subscribeToPushNotifications',
```
Comment (Prithpal-Sooriya) [inline app/scripts/metamask-controller.js:1568]: Nice, this is the fix for the silent push notifications!

File: test/e2e/tests/notifications/mocks.ts [hunk 3]
```diff
@@ -24,15 +24,15 @@ type MockResponse = {
  *
  * @param server - server obj used to mock our endpoints
  */
-export function mockNotificationServices(server: Mockttp) {
+export async function mockNotificationServices(server: Mockttp) {
```
Comment (Prithpal-Sooriya) [inline test/e2e/tests/notifications/mocks.ts:27]: Ensure that we update the e2e setup file that uses this.

Comment (Prithpal-Sooriya) [reply test/e2e/tests/notifications/mocks.ts:27]: `test/e2e/mock-e2e.js`

**Review by Prithpal-Sooriya — DISMISSED** [review]

**Review by mathieuartu — DISMISSED** [review]

**Review by Prithpal-Sooriya — APPROVED** [review]

**Review by mathieuartu — APPROVED** [review]

**Pull request merged**